internal/getproviders/package_authentication.go

```diff
@@ -491,7 +487,13 @@ func (s signatureAuthentication) findSigningKey() (*SigningKey, string, error) {
 			return nil, "", fmt.Errorf("error decoding signing key: %w", err)
 		}
 
-		entity, err := openpgp.CheckDetachedSignature(keyring, bytes.NewReader(s.Document), bytes.NewReader(s.Signature), openpgpConfig)
+		entity, err := openpgp.CheckDetachedSignature(keyring, bytes.NewReader(s.Document), bytes.NewReader(s.Signature), nil)
+		if err == openpgpErrors.ErrSignatureExpired || err == openpgpErrors.ErrKeyExpired {
+			// Internally openpgp will *only* return the Expired errors if all other checks have succeded
+			// This is currently the best way to work around expired provider keys
+			fmt.Printf("[WARN] Provider %s/%s (%v) gpg key expired, this will fail in future versions of OpenTofu: '%s'\n", s.Meta.Provider.Namespace, s.Meta.Provider.Type, s.Meta.Provider.Hostname, s.Meta.Location.String())
```

Comment:
I think we don't need to print the location here, Printing the provider name is enough is enough.